superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterValue.tsx

```diff
@@ -76,12 +80,28 @@ const FilterValue: React.FC<FilterProps> = ({
         requestParams: { dashboardId: 0 },
       })
         .then(response => {
-          setState(response.result);
-          setError(false);
-          setLoading(false);
+          if (isFeatureEnabled(FeatureFlag.GLOBAL_ASYNC_QUERIES)) {
+            // deal with getChartDataRequest transforming the response data
+            const result = 'result' in response ? response.result[0] : response;
```

Comment:
I propose doing it in a follow up: In addition to cleaning up `getChartDataRequest`, I think it would be a good idea to add typing to some of these files. We may also want to consider moving the async handling logic into `getChartDataRequest`, so that the consumers don't need to be aware of whether we're using sync or async retrieval.